Generating code from protocol buffers definitions requires the executable "protoc-gen-gocosmos"

I ran `make proto-gen` and got this 

```
$ make proto-gen
GO111MODULE=on go mod tidy
go mod vendor
vendoring non-go files...
/home/ericu/akash/akash/.cache/bin/modvendor -copy="**/*.proto" -include=\
github.com/cosmos/cosmos-sdk/proto,\
github.com/cosmos/cosmos-sdk/third_party/proto
/home/ericu/akash/akash/.cache/bin/modvendor -copy="**/*.h **/*.c" -include=\
github.com/zondax/hid
/home/ericu/akash/akash/.cache/bin/modvendor -copy="**/swagger.yaml" -include=\
github.com/cosmos/cosmos-sdk/client/docs
docker run --rm -v /home/ericu/akash/akash:/workspace -w /workspace tendermintdev/docker-build-proto find ./ ! -path "./vendor/*" -name *.proto -exec clang-format -i {} \;
./script/protocgen.sh
protoc-gen-gocosmos: program not found or is not executable
Please specify a program using absolute path or make sure the program is available in your PATH system variable
--gocosmos_out: protoc-gen-gocosmos: Plugin failed with status code 1.
make/codegen.mk:24: recipe for target 'proto-gen' failed
make: *** [proto-gen] Error 1

```

It looks like we are dependent on this executable that @troian  found 

https://github.com/regen-network/cosmos-proto/tree/master/protoc-gen-gocosmos

I'm not entirely sure what role this program plays. We need to at a minimum make acquiring this executable part of the Makefile target

--- a/Makefile
+++ b/Makefile
@@ -16,20 +16,23 @@ CACHE_BIN             := $(CACHE)/bin
 CACHE_INCLUDE         := $(CACHE)/include
 CACHE_VERSIONS        := $(CACHE)/versions
 
-BUF_VERSION           ?= 0.25.0
-PROTOC_VERSION        ?= 3.13.0
-GRPC_GATEWAY_VERSION  ?= 1.14.7
-GOLANGCI_LINT_VERSION ?= v1.31.0
-GOLANG_VERSION        ?= 1.15.2
-GOLANG_CROSS_VERSION  := v$(GOLANG_VERSION)
+BUF_VERSION                ?= 0.25.0
+PROTOC_VERSION             ?= 3.13.0
+PROTOC_GEN_COSMOS_VERSION  ?= master
+GRPC_GATEWAY_VERSION       ?= 1.14.7
+GOLANGCI_LINT_VERSION      ?= v1.31.0
+GOLANG_VERSION             ?= 1.15.2
+GOLANG_CROSS_VERSION       := v$(GOLANG_VERSION)
 
 # <TOOL>_VERSION_FILE points to the marker file for the installed version.
 # If <TOOL>_VERSION_FILE is changed, the binary will be re-downloaded.
-PROTOC_VERSION_FILE        = $(CACHE_VERSIONS)/protoc/$(PROTOC_VERSION)
-GRPC_GATEWAY_VERSION_FILE  = $(CACHE_VERSIONS)/protoc-gen-grpc-gateway/$(GRPC_GATEWAY_VERSION)
-MODVENDOR                  = $(CACHE_BIN)/modvendor
-PROTOC                    := $(CACHE_BIN)/protoc
-GRPC_GATEWAY              := $(CACHE_BIN)/protoc-gen-grpc-gateway
+PROTOC_VERSION_FILE             = $(CACHE_VERSIONS)/protoc/$(PROTOC_VERSION)
+GRPC_GATEWAY_VERSION_FILE       = $(CACHE_VERSIONS)/protoc-gen-grpc-gateway/$(GRPC_GATEWAY_VERSION)
+PROTOC_GEN_COSMOS_VERSION_FILE  = $(CACHE_VERSIONS)/protoc-gen-cosmos/$(PROTOC_GEN_COSMOS_VERSION)
+MODVENDOR                       = $(CACHE_BIN)/modvendor
+PROTOC                         := $(CACHE_BIN)/protoc
+PROTOC_GEN_COSMOS              := $(CACHE_BIN)/protoc-gen-cosmos
+GRPC_GATEWAY                   := $(CACHE_BIN)/protoc-gen-grpc-gateway
 
 DOCKER_RUN            := docker run --rm -v $(shell pwd):/workspace -w /workspace
 DOCKER_BUF            := $(DOCKER_RUN) bufbuild/buf:$(BUF_VERSION)

--- a/make/codegen.mk
+++ b/make/codegen.mk
@@ -19,7 +19,7 @@ kubetypes: deps-vendor
 	akash.network:v1
 
 .PHONY: proto-gen
-proto-gen: $(PROTOC) $(GRPC_GATEWAY) modvendor proto-format
+proto-gen: $(PROTOC) $(GRPC_GATEWAY) $(PROTOC_GEN_COSMOS) modvendor proto-format
 	./script/protocgen.sh
 
 .PHONY: proto-swagger-gen

--- a/make/setup-cache.mk
+++ b/make/setup-cache.mk
@@ -18,6 +18,15 @@ $(PROTOC_VERSION_FILE): $(CACHE)
 	touch $@
 $(PROTOC): $(PROTOC_VERSION_FILE)
 
+$(PROTOC_GEN_COSMOS_VERSION_FILE): $(CACHE)
+	@echo "installing protoc-gen-cosmos..."
+	rm -f $(PROTOC_GEN_COSMOS)
+	GOBIN=$(CACHE_BIN) $(GO) install github.com/regen-network/cosmos-proto/protoc-gen-gocosmos
+	rm -rf "$(dir $@)"
+	mkdir -p "$(dir $@)"
+	touch $@
+$(PROTOC_GEN_COSMOS): $(PROTOC_GEN_COSMOS_VERSION_FILE)
+
 $(GRPC_GATEWAY_VERSION_FILE): $(CACHE)
 	@echo "Installing protoc-gen-grpc-gateway..."
 	rm -f $(GRPC_GATEWAY)

--- a/script/protocgen.sh
+++ b/script/protocgen.sh
@@ -23,7 +23,6 @@ Mgoogle/protobuf/any.proto=github.com/cosmos/cosmos-sdk/codec/types:. \
   -I "vendor/github.com/cosmos/cosmos-sdk/third_party/proto" \
   --grpc-gateway_out=logtostderr=true:. \
   $(find "${dir}" -maxdepth 1 -name '*.proto')
-
 done
 
 # move proto files to the right places

